File: backend/internal/domain/stock.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Stock struct {
	ID         uuid.UUID `json:"id"`
	Ticker     string    `json:"ticker"`
	Company    string    `json:"company"`
	Brokerage  string    `json:"brokerage"`
	Action     string    `json:"action"`
	RatingFrom string    `json:"ratingFrom"`
	RatingTo   string    `json:"ratingTo"`
	TargetFrom float64   `json:"targetFrom"`
	TargetTo   float64   `json:"targetTo"`
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`
}

type StockFilter struct {
	Search    string
	Ticker    string
	Action    string
	SortBy    string
	SortOrder string
	Page      int
	Limit     int
}

func NewStockFilter() StockFilter {
	return StockFilter{
		SortBy:    "created_at",
		SortOrder: "desc",
		Page:      1,
		Limit:     20,
	}
}

type StockRecommendation struct {
	Stock           Stock    `json:"stock"`
	Score           float64  `json:"score"`
	Reasons         []string `json:"reasons"`
	UpsidePotential float64  `json:"upsidePotential"`
}

type PaginatedStocks struct {
	Data       []Stock `json:"data"`
	Page       int     `json:"page"`
	Limit      int     `json:"limit"`
	TotalCount int64   `json:"totalCount"`
	TotalPages int     `json:"totalPages"`
	HasNext    bool    `json:"hasNext"`
	HasPrev    bool    `json:"hasPrev"`
}

File: frontend/src/types/stock.ts
export interface Stock {
  id: string
  ticker: string
  company: string
  brokerage: string
  action: string
  ratingFrom: string
  ratingTo: string
  targetFrom: number
  targetTo: number
  createdAt: string
  updatedAt: string
}

export interface StockFilter {
  search: string
  sortBy: SortField
  sortOrder: SortOrder
  action: string
  page: number
  limit: number
}

export type SortField = 'ticker' | 'company' | 'action' | 'targetTo' | 'createdAt'
export type SortOrder = 'asc' | 'desc'

export interface StockRecommendation {
  stock: Stock
  score: number
  reasons: string[]
  upsidePotential: number
}

export interface PaginatedResponse<T> {
  data: T[]
  page: number
  limit: number
  totalCount: number
  totalPages: number
  hasNext: boolean
  hasPrev: boolean
}

export const defaultFilter: StockFilter = {
  search: '',
  sortBy: 'createdAt',
  sortOrder: 'desc',
  action: '',
  page: 1,
  limit: 20
}
